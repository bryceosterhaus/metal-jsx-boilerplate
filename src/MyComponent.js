import Component, {Config} from 'metal-jsx';

class MyComponent extends Component {
	render() {
		return <div>{name}</div>;
	}
}

MyComponent.PROPS = {
	name: Config.string().value('MyComponent')
};

MyComponent.STATE = {};

export default MyComponent;
