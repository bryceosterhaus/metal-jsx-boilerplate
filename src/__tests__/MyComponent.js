import MyComponent from '../MyComponent';

describe('MyComponent', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('renders', () => {
		component = new MyComponent();

		expect(component).toMatchSnapshot();
	});
});
