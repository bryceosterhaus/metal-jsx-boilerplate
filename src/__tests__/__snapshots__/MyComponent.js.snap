// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent renders 1`] = `<div>nodejs</div>`;
